include/sycl_ext_complex.hpp

```diff
@@ -329,6 +333,33 @@ template <class _A1> class __promote_imp<_A1, void, void, true> {
 
 template <class _A1, class _A2 = void, class _A3 = void>
 class __promote : public __promote_imp<_A1, _A2, _A3> {};
+
```

Comment:
Can you put a comment here, to said that it's a standard "fast math optimization" that compiler already does, but for whatever reason the `isnan / finate` branches are not optimized away?